src/MeshKernel/CurvilinearGridLineShift.cpp

```diff
@@ -0,0 +1,180 @@
+//---- GPL ---------------------------------------------------------------------
+//
+// Copyright (C)  Stichting Deltares, 2011-2021.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+// contact: delft3d.support@deltares.nl
+// Stichting Deltares
+// P.O. Box 177
+// 2600 MH Delft, The Netherlands
+//
+// All indications and logos of, and references to, "Delft3D" and "Deltares"
+// are registered trademarks of Stichting Deltares, and remain the property of
+// Stichting Deltares. All rights reserved.
+//
+//------------------------------------------------------------------------------
+
+#include <MeshKernel/CurvilinearGrid.hpp>
+#include <MeshKernel/CurvilinearGridLine.hpp>
+#include <MeshKernel/CurvilinearGridLineShift.hpp>
+#include <MeshKernel/Entities.hpp>
+
+using meshkernel::CurvilinearGrid;
+using meshkernel::CurvilinearGridLineShift;
+using meshkernel::Point;
+
+CurvilinearGridLineShift::CurvilinearGridLineShift(std::shared_ptr<CurvilinearGrid> grid) : CurvilinearGridAlgorithm(grid)
+
+{
+    // Compute the grid node types
+    m_gridModified = std::make_shared<CurvilinearGrid>(m_grid->m_gridNodes, m_grid->m_projection);
+}
+
+std::shared_ptr<CurvilinearGrid> CurvilinearGridLineShift::Compute()
+{
+    if (m_lines.empty())
+    {
+        throw std::invalid_argument("CurvilinearGridLineShift::Compute No candidate line to shift has been selected");
+    }
+
+    /// The first delta is
+    auto const previousNodeIndex = m_lines[0].m_startNode;
+    auto previousDelta = m_gridModified->m_gridNodes[previousNodeIndex.m_m][previousNodeIndex.m_n] -
+                         m_grid->m_gridNodes[previousNodeIndex.m_m][previousNodeIndex.m_n];
+
+    const double eps = 1e-5;
+    auto previousCoordinate = m_lines[0].m_startCoordinate;
+    for (auto i = 1; i <= m_lines[0].m_endCoordinate; ++i)
+    {
+        auto const currentNodeIndex = m_lines[0].GetNodeindexFromCoordinate(i);
+
+        auto const currentDelta = m_gridModified->m_gridNodes[currentNodeIndex.m_m][currentNodeIndex.m_n] -
```

Comment:
feel free to ignore this comment, I am in "stream of conciousness" mode. 
the type of currentDelta is "Point" but it could also be seen as a 2d vector. If it were a vector, then the check below could be made easier, because then you coul check currentdDelta.Norm() < eps instead of checking both components in turn.